ds1307/ds1307.go

```diff
@@ -0,0 +1,145 @@
+package ds1307
+
+import (
+	"time"
+
+	"machine"
+)
+
+// Device wraps an I2C connection to a DS1307 device.
+type Device struct {
+	bus        machine.I2C
+	address    uint8
```

Comment:
We have been making the `Address` member of the struct public, so that if someone runs into a weird device with a non-standard I2C address, they can override it.